apps/cli-gui/src/main.ts

```diff
@@ -1,32 +1,18 @@
-import { HttpClient, provideHttpClient } from '@angular/common/http';
-import { APP_INITIALIZER } from '@angular/core';
+import { provideHttpClient } from '@angular/common/http';
 import { bootstrapApplication } from '@angular/platform-browser';
 import { provideAnimations } from '@angular/platform-browser/animations';
-import { provideRouter, withEnabledBlockingInitialNavigation } from '@angular/router';
-import { map } from 'rxjs';
+import {
+  provideRouter,
+  withEnabledBlockingInitialNavigation,
+} from '@angular/router';
 
 import { AppComponent } from './app/app.component';
 import { APP_ROUTES } from './app/app.routes';
-import { CoreService } from './app/core/core.service';
 
 bootstrapApplication(AppComponent, {
   providers: [
     provideRouter(APP_ROUTES, withEnabledBlockingInitialNavigation()),
     provideHttpClient(),
     provideAnimations(),
-    {
-      provide: APP_INITIALIZER,
-      useFactory: (http: HttpClient, core: CoreService) => () =>
-        http.get<string[]>(`/api/workspace`).pipe(
```

Comment:
Do we not need the list of projects and the current project? How do you plan to run generators?

I thought we'd keep the last workspace path in LS, so if we have one, we connect to it, and there is no need to re-connect on every page reload.

Reply:
Right- added this logic in the guard

Reply:
use sessionStorage instead, so you won't have to deal with invalidation. its also better UX IMO, the data will be persistent until the tab is closed (which should be the same as terminating the gui process)

Reply:
done